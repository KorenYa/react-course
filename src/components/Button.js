import React, { PureComponent } from 'react';
class Button extends PureComponent {
    render() {
        return <button>Push me</button>;
    }
}
export default Button;
