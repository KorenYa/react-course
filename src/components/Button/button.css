.btn {
    display: inline-block;
    font-weight: 600;
    color: #ffffff;
    text-align: center;
    text-transform: uppercase;
    vertical-align: middle;
    user-select: none;
    background-color: #3c3c3c;
    font-size: 10px;
    padding: 4px 12px;
    line-height: 1.5;
    border-radius: 2px;
    border: none;
    margin-left: 10px;
}

.btn:hover {
    cursor: pointer;
}

.btn-lg {
    font-size: 14px;
    padding: 8px 24px;
}

.btn-back {
    color: #f55263;
    background-color: #ffffff;
}

.btn-back:hover {
    color: #ffffff;
}

.btn:hover, .btn.active {
    background-color: #f55263;
}