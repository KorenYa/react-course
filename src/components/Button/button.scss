@import '../../styles/constants.scss';

.btn {
    display: inline-block;
    font-weight: $bold;
    color: $white;
    text-align: center;
    text-transform: uppercase;
    vertical-align: middle;
    user-select: none;
    background-color: $grey;
    font-size: $sm;
    padding: 4px 12px;
    line-height: 1.5;
    border-radius: 2px;
    border: none;
    margin-left: 10px;

    &-lg {
        font-size: $md;
        padding: 8px 24px;
    }

    &:hover {
        cursor: pointer;
    }

    &:hover,
    &.active {
        background-color: $red;
    }
}
