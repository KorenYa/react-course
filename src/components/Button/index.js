import React from 'react';
import PropTypes from 'prop-types';
import './button.css';
const Button = function(props) {
    return (
        <button
            onClick={() => props.onClick(event, props.searchBy)}
            className={`btn ${props.styleName}`}
            type={props.type || ''}
            data-searchby={props.searchBy}
        >
            {props.text}
        </button>
    );
};
Button.propTypes = {
    type: PropTypes.string,
    text: PropTypes.string,
    searchBy: PropTypes.string,
    styleName: PropTypes.string,
    onClick: PropTypes.func
};
export default Button;
