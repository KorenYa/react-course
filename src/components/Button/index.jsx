// @flow
import React from 'react';
import './button.scss';

type ButtonProps = {
  type: string,
  text: string,
  searchBy: string,
  styleName: string,
  onClick: func,
};

const Button = (props: ButtonProps) => {
  const { onClick, searchBy, styleName, type, text } = props;

  return (
    <button
      onClick={() => onClick(event, searchBy)}
      className={`btn ${styleName}`}
      type={type || ''}
      data-searchby={searchBy}
    >
      {text}
    </button>
  );
};

export default Button;
