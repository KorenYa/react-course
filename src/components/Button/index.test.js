import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';

describe('Button', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(
                <Button
                    type='button'
                    text='click me'
                    seachBy='genre'
                    styleName='btn-lg'
                    onClick='handler'
                />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
