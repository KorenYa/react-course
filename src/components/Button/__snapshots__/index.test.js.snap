// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="btn btn-lg"
  onClick={[Function]}
  type="button"
>
  click me
</button>
`;
